--- nfs-ganesha-4.1/src/cmake/modules/FindNTIRPC.cmake.orig	2022-11-18 19:39:44.000000000 +0000
+++ nfs-ganesha-4.1/src/cmake/modules/FindNTIRPC.cmake	2022-11-21 14:31:09.148000000 +0000
@@ -67,15 +67,18 @@
 
 if (NOT NTIRPC_LIBRARY_DIR)
 	find_path(NTIRPC_LIBRARY_DIR
-    NAMES libntirpc.so
+    		NAMES libntirpc.so
 		PATHS ${NTIRPC_PREFIX}
-    PATH_SUFFIXES lib/${CMAKE_LIBRARY_ARCHITECTURE} lib lib64
+    		PATH_SUFFIXES lib/${CMAKE_LIBRARY_ARCHITECTURE} lib lib64
 		DOC "The NTIRPC libraries")
 endif (NOT NTIRPC_LIBRARY_DIR)
 
-find_library(NTIRPC_LIBRARY ntirpc PATHS ${NTIRPC_LIBRARY_DIR} NO_DEFAULT_PATH)
-find_library(NTIRPC_TRACEPOINTS ntirpc_tracepoints PATHS ${NTIRPC_LIBRARY_DIR} NO_DEFAULT_PATH)
-find_library(NTIRPC_LTTNG ntirpc_lttng PATHS ${NTIRPC_LIBRARY_DIR} NO_DEFAULT_PATH)
+find_library(NTIRPC_LIBRARY ntirpc
+		PATHS_SUFFIXES ${CMAKE_LIBRARY_ARCHITECTURE})
+find_library(NTIRPC_TRACEPOINTS ntirpc_tracepoints
+		PATHS_SUFFIXES ${CMAKE_LIBRARY_ARCHITECTURE})
+find_library(NTIRPC_LTTNG ntirpc_lttng
+		PATHS_SUFFIXES ${CMAKE_LIBRARY_ARCHITECTURE})
 
 set(NTIRPC_VERSION_HEADER "${NTIRPC_INCLUDE_DIR}/version.h")
 if (EXISTS ${NTIRPC_VERSION_HEADER})
--- nfs-ganesha-4.1/src/scripts/gpfs-epoch/CMakeLists.txt.orig	2022-11-18 19:39:44.000000000 +0000
+++ nfs-ganesha-4.1/src/scripts/gpfs-epoch/CMakeLists.txt	2022-11-21 14:31:09.152000000 +0000
@@ -21,7 +21,7 @@
 #-------------------------------------------------------------------------------
 
 if(USE_FSAL_GPFS)
-  if(Python_FOUND)
+  if(PYTHONINTERP_FOUND)
     set(SETUP_PY_IN "${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in")
     set(SETUP_PY    "${CMAKE_CURRENT_BINARY_DIR}/setup.py")
     set(OUTPUT      "${CMAKE_CURRENT_BINARY_DIR}/build/gpfs_epoch_timestamp")
@@ -59,7 +59,7 @@
     add_custom_command(
       OUTPUT ${OUTPUT}
       COMMAND ${CMAKE_COMMAND} -E touch __init__.py
-      COMMAND ${Python_EXECUTABLE} "${SETUP_PY}"  build
+      COMMAND ${PYTHON_EXECUTABLE} "${SETUP_PY}"  build
       COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
       DEPENDS ${GPFS_EPOCH_SRCS} ${SCRIPTS}
       )
@@ -68,7 +68,7 @@
 
     install(
       CODE
-      "execute_process(WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} COMMAND ${Python_EXECUTABLE} ${SETUP_PY} install --skip-build --no-compile --prefix=\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX} --install-scripts=\$ENV{DESTDIR}${LIBEXECDIR}/ganesha)"
+      "execute_process(WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install --skip-build --no-compile --prefix=\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX} --install-scripts=\$ENV{DESTDIR}${LIBEXECDIR}/ganesha)"
       )
-  endif(Python_FOUND)
+  endif(PYTHONINTERP_FOUND)
 endif(USE_FSAL_GPFS)
--- nfs-ganesha-4.1/src/scripts/ganeshactl/CMakeLists.txt.orig	2022-11-18 19:39:44.000000000 +0000
+++ nfs-ganesha-4.1/src/scripts/ganeshactl/CMakeLists.txt	2022-11-21 14:31:09.152000000 +0000
@@ -139,7 +139,7 @@
 
   add_custom_command(
     OUTPUT ${OUTPUT}
-    COMMAND ${Python_EXECUTABLE} "${SETUP_PY}"  build
+    COMMAND ${PYTHON_EXECUTABLE} "${SETUP_PY}"  build
     COMMAND ${CMAKE_COMMAND} -E touch ${OUTPUT}
     DEPENDS ${GANESHA_SRCS} ${UI_PY_FILES} ${SCRIPTS}
     )
@@ -148,9 +148,9 @@
 
   install(
     CODE
-    "execute_process(WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} COMMAND ${Python_EXECUTABLE} ${SETUP_PY} install --skip-build --no-compile --prefix=\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX})"
+    "execute_process(WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR} COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install --skip-build --no-compile --prefix=\$ENV{DESTDIR}${CMAKE_INSTALL_PREFIX})"
     )
-endif(Python_FOUND)
+endif(PYTHONINTERP_FOUND)
 
 # Man page
 set(man8_file ganesha_conf.man)
--- nfs-ganesha-4.1/src/CMakeLists.txt.orig	2022-11-18 19:39:44.000000000 +0000
+++ nfs-ganesha-4.1/src/CMakeLists.txt	2022-11-21 14:31:09.152000000 +0000
@@ -578,13 +578,19 @@
 
 gopt_test(USE_ADMIN_TOOLS)
 if (USE_ADMIN_TOOLS)
-  find_package (Python COMPONENTS Interpreter Development)
-  if (NOT Python_Interpreter_FOUND)
-    message(WARNING "Cannot find python. Disablin admin tools")
+  find_package(PythonInterp ${USE_ADMIN_TOOLS_REQUIRED})
+  if (NOT PYTHONINTERP_FOUND)
+    message(WARNING "Cannot find python. Disabling admin tools")
     set(USE_ADMIN_TOOLS OFF)
-  endif (NOT Python_Interpreter_FOUND)
+  endif (NOT PYTHONINTERP_FOUND)
 endif (USE_ADMIN_TOOLS)
 
+get_cmake_property(_variableNames VARIABLES)
+list(SORT _variableNames)
+foreach(_variableName ${_variableNames})
+    message(STATUS "${_variableName}=${${_variableName}}")
+endforeach()
+
 gopt_test(USE_GUI_ADMIN_TOOLS)
 if (USE_ADMIN_TOOLS)
   if (USE_GUI_ADMIN_TOOLS)
@@ -1184,6 +1190,7 @@
 
 set(NTIRPC_MIN_VERSION 4.0)
 if (USE_SYSTEM_NTIRPC)
+  set(NTIRPC_INCLUDE_DIR "/usr/include/ntirpc/")
   find_package(NTIRPC ${NTIRPC_MIN_VERSION} REQUIRED)
   if (USE_LTTNG)
     set(NTIRPC_LIBRARY ${NTIRPC_LIBRARY} ${NTIRPC_LTTNG})
